### PR TITLE
converts add-metadata to be click-based

### PR DESCRIPTION
**Pull request by gregcaporaso**

### CODE REVIEW

File: biom/cli/metadata_adder.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                                                process_fns=process_fns,
+                                                header=sample_header)
+
+    if observation_metadata is not None:
+        observation_metadata = MetadataMap.from_file(
+            observation_metadata,
+            process_fns=process_fns,
+            header=observation_header)
+
+    # NAUGHTY: this is modifying the input table IN PLACE!!! And then
```
Comment (wasade) [inline biom/cli/metadata_adder.py]: yes should stay without question

File: scripts/biomcli [hunk 2]
```diff
@@ -51,6 +53,107 @@ def summarize_table(input_fp, output_fp, qualitative, observations):
...
+    examples here: http://biom-format.org/documentation/adding_metadata.html
+
+    Example usage:
+
+    Add sample metadata to a BIOM table:
+
+    $ biom add-metadata -i otu_table.biom -o table_with_sample_metadata.biom
+      -m sample_metadata.txt
+    """
+    table = biom.parse.load_table(input_fp)
```
Comment (wasade) [inline scripts/biomcli]: `table = biom.load_table`

File: tests/test_cli/test_add_metadata.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+# -----------------------------------------------------------------------------
+# Copyright (c) 2011-2015, The BIOM Format Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from biom.cli import add_metadata
+from biom.parse import load_table
```
Comment (wasade) [inline tests/test_cli/test_add_metadata.py]: `from biom import load_table`

...but, doesn't really matter for the time being

**Comment by wasade** [comment]
minor comment, :+1: otherwise

**Comment by gregcaporaso** [comment]
ok, updated those. should be ready for merge (probably don't need to let the tests re-run if you want to get access to ``write_biom_table``, since they'll run again in a few mins for a different PR.

**Pull request merged**